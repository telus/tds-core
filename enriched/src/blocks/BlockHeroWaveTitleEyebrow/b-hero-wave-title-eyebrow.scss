@import '~telus-thorium-core/scss/settings/variables';

.b-hero-wave-title__bg {
  background-image: url('./wave_header_default.png');
  background-position: top left;
  background-repeat: repeat-x;
  padding-top: 105px;
  margin-bottom: 48px;
  background-size: contain;
}

.b-hero-wave-title--grape {
  color: $color-grape;
}
