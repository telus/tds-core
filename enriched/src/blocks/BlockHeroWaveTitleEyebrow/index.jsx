import React from 'react';

if ( process.env.BROWSER ) {
  require('./b-hero-wave-title-eyebrow.scss');
}

const BlockHeroWaveTitleEyebrow = (props) => {
  const { eyebrow, title } = props;

  return (
    <div className="b-hero-wave-title-eyebrow__bg">
      <div className="container container--limited-width">
        <div className="grid-row">
          <div className="medium-6 large-7 xl-7">
            <p className="heading-4 b-hero-wave-title-eyebrow--grape">{eyebrow}</p>
            <h1 className="display-heading-1">{title}</h1>
          </div>
        </div>
      </div>
    </div>
  );
};

BlockHeroWaveTitleEyebrow.propTypes = {
  title: React.PropTypes.string.isRequired,
  eyebrow: React.PropTypes.string
};

BlockHeroWaveTitleEyebrow.defaultProps = {
  eyebrow: ''
};

export default BlockHeroWaveTitleEyebrow;
