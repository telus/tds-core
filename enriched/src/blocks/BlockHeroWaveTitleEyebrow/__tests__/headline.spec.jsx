import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import BlockHeroWaveTitleEyebrow from '../';

describe('BlockHeroWaveTitleEyebrow', () => {
  const data = { eyebrow: 'Foo', title: 'Bar' };

  it('matches snapshot', () => {
    const wrapper = mount(<BlockHeroWaveTitleEyebrow {...data} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render eyebrow', () => {
    const wrapper = mount(<BlockHeroWaveTitleEyebrow {...data} />);
    expect(wrapper.find('p').first().render().text()).toEqual('Foo');
  });

  it('should render without subcategory', () => {
    const wrapper = mount(<BlockHeroWaveTitleEyebrow title={data.title} />);
    expect(wrapper.find('p').first().render().text()).toEqual('');
  });

  it('should render title', () => {
    const wrapper = mount(<BlockHeroWaveTitleEyebrow {...data} />);
    expect(wrapper.find('h1').first().render().text()).toEqual('Bar');
  });

  it('should use headline style', () => {
    const wrapper = mount(<BlockHeroWaveTitleEyebrow {...data} />);
    expect(wrapper.find('div').first().hasClass('b-hero-wave-title-eyebrow__bg')).toBeTruthy();
  });
});
