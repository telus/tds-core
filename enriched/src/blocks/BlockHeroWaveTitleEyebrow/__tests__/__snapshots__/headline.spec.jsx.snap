exports[`BlockHeroWaveTitleEyebrow matches snapshot 1`] = `
<BlockHeroWaveTitleEyebrow
  eyebrow="Foo"
  title="Bar">
  <div
    className="b-hero-wave-title-eyebrow__bg">
    <div
      className="container container--limited-width">
      <div
        className="grid-row">
        <div
          className="medium-6 large-7 xl-7">
          <p
            className="heading-4 b-hero-wave-title-eyebrow--grape">
            Foo
          </p>
          <h1
            className="display-heading-1">
            Bar
          </h1>
        </div>
      </div>
    </div>
  </div>
</BlockHeroWaveTitleEyebrow>
`;
